v2/src/log_sink_etw.c

```diff
@@ -0,0 +1,497 @@
+                    *pos = TlgInINT8;
+                    break;
+                case LOG_CONTEXT_PROPERTY_TYPE_uint8_t:
+                    /* Codes_SRS_LOG_SINK_ETW_01_079: [ If the property type is LOG_CONTEXT_PROPERTY_TYPE_uint8_t, a byte with the value TlgInUINT8 shall be added in the metadata. ]*/
+                    *pos = TlgInUINT8;
+                    break;
+
+                case LOG_CONTEXT_PROPERTY_TYPE_struct:
+                    /* Codes_SRS_LOG_SINK_ETW_01_052: [ For struct properties one extra byte shall be added for the field count. ]*/
+                    /* Codes_SRS_LOG_SINK_ETW_01_056: [ If the property type is LOG_CONTEXT_PROPERTY_TYPE_struct, a byte with the value _TlgInSTRUCT | _TlgInChain shall be added in the metadata. ]*/
+                    *pos = _TlgInSTRUCT | _TlgInChain;
+                    pos++;
+
+                    if (pos == &self_described_event->metadata[MAX_METADATA_SIZE])
+                    {
+                        add_properties = false;
+                    }
+                    else
+                    {
+                        /* Codes_SRS_LOG_SINK_ETW_01_057: [ If the property is a struct, an extra byte shall be added in the metadata containing the number of fields in the structure. ]*/
+                        *pos = *((uint8_t*)context_property_value_pairs[i].value);
+                    }
+                    break;
+                }
+
+                if (!add_properties)
+                {
+                    break;
+                }
+                else
+                {
+                    pos++;
+                }
+            }
+        }
+        else
+        {
+            /* Codes_SRS_LOG_SINK_ETW_01_082: [ If more than 64 properties are given in log_context, log_sink_etw.log_sink_log shall not add any properties to the event. ]*/
+            add_properties = false;
+        }
+
+        /* Codes_SRS_LOG_SINK_ETW_01_026: [ log_sink_etw.log_sink_log shall fill a SELF_DESCRIBED_EVENT structure, setting the following fields: ]*/
+        uint16_t metadata_size;
+
+        if (!add_properties)
+        {
+            metadata_size = metadata_size_without_properties;
+            pos = &self_described_event->metadata[metadata_size_without_properties];
+        }
+        else
+        {
+            metadata_size = (uint16_t)(pos - &self_described_event->metadata[0]);
+        }
+
+        char* formatted_message = (char*)pos;
+        size_t available_bytes = sizeof(_tlgEvent) - (formatted_message - (char*)self_described_event);
+
+        int formatted_message_length = vsnprintf(formatted_message, available_bytes, message, args);
+        if (formatted_message_length < 0)
+        {
+            /* Codes_SRS_LOG_SINK_ETW_01_086: [ If any error occurs log_sink_etw.log_sink_log shall print Error emitting ETW event and return. ]*/
+            (void)printf("Error emitting ETW event.\r\n");
```

Comment:
>event [](http://example.com/codeflow?start=45&length=5)

maybe there's some enrichment we can have there as to add some strings or more information that would help tracking "which/what" event failed.

Reply:
Added